src/main/java/org/entando/plugins/pda/pam/service/task/KieTaskService.java

```diff
@@ -101,6 +103,16 @@ public Task get(Connection connection, AuthenticatedUser user, String id) {
         return task;
     }
 
+    @Override
+    public SimpleRestResponse<Set<String>> listTaskColumns(Connection connection, AuthenticatedUser user) {
+        PagedListRequest pagedRequest = new PagedListRequest(1, 1, "taskId", PagedListRequest.DIRECTION_VALUE_DEFAULT);
+        PagedRestResponse<Task> list = list(connection, user, pagedRequest);
+        if (list.getPayload().isEmpty()) {
+            return new SimpleRestResponse<>(Collections.emptySet());
+        }
+        return new SimpleRestResponse<>(list.getPayload().get(0).getData().keySet());
+    }
+
```

Comment:
This is not correct.
Every Task may be different from each other.

Perhaps we should be using the ProcessDefinition and TaskDefinition endpoints.

Reply:
@ffleandro This is exact the same point a raised in a daily some time ago. At that moment, it was mentioned that I shouldn't worry about it and get the data struct from the first task on the task list. I even mentioned that if we don't have tasks, I don't have data to return. Indeed this is also how it is implemented in the legacy source code. So, I did this implementation based on that discussion and the legacy implementation that we have right now. The only change I think we should do here is to return the basic fields in case we don't have any tasks in that list. We can discuss new ideas in a second interaction

Reply:
Ok, let's investigate further if we can access the Task Definition instead of querying the first task, because each task can be different and have different columns.